add core developer document

syuhei176 opened the pull request.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+OVM is the concept to standardize L2 dispute logic. L2 developers write property to define dispute logic for Layer 2 protocol.
+Every properties are claimed to the same Contract called Adjudication Contract. Properties are written by predicate logic and it can be prove to true or false under dispute logic.
+For example, users deposit their asset to Deposit Contract and finally withdraw asset from Deposit Contract. In this scenario, Property stands for the condition which users can withdraw money from Deposit Contract.
+In this document, we describe how we can implement OVM contract.
```
later in this document, we will describe how you can implement this kind of contract on the OVM.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### Property
+
+Property stadns for dispute logic and we can claim every Properties to Adjudicator Contract. Property has its predicate address and array of input.
```
Property stands for dispute logic and we can claim any kind of Properties to the Adjudicator Contract. Property has its predicate address and array of inputs.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+### Decider and Quantifier
+
+We are using quantifier elimination technich to make OVM contract imeplementation simple.
+There are 2 rules. So evetything is decider in our implementation.
```
We use the quantifier elimination technique to keep our OVM contract implementation simple.
There are 2 rules. So evetything is decider in our implementation.
> So evetything is decider in our implementation. <=ここの意味が？

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### claimProperty
+
+Claims property and instantiate new challenge game.
```
Claims property and instantiates a new challenge game.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### isDecided
+
+Gets the claim is decided or not.
```
Gets the result of whether the claim has been decided or not.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### decideClaimToTrue
+
+Set the game decision true whose dispute period has passed.
```
Sets the game decision true whose dispute period has already passed.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### decideClaimToFalse
+
+Set the game decision false when its challenge is set true.
```
Set the game decision false when its challenge has been evaluated to true.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### removeChallenge
+
+Removes challenge whose decision is set as false.
```
Removes challenge whose decision has been evaluated to false.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### setPredicateDecision
+
+This is called from Predicate. It sets a game decision. For example, you know signature of `IsValidSignature(message, address, signature)`, you can set the game decision true which has this claim through `IsValidSignature` predicate.
```
This method is called by Predicate. It sets a game decision. For example, if you know the signature of `IsValidSignature(message, address, signature)`, you can set the game decision true which has this claim through `IsValidSignature` predicate.
>このwhichがどこにかかるのかが？

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### challenge
+
+Adds new challenge game. AC must check the challenge is valid of parent game.
```
Adds a new challenge game. The Adjudication Contract has to check if the challenge is valid in the parent game.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+isValidChallenge validates valid child node of game tree. I explain game tree.
+
+I write the "challenge rule table" defining `challenge` function which applies the challenge table below recursively.
+For example, `challenge(¬p) = p`.
```
`isValidChallenge` validates valid child node of a game tree. I explain a game tree.
Game tree: 
You write the "challenge rule table" defining `challenge` function which applies the challenge table below recursively.
For example, `challenge(¬p) = p`.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+#### Why compile?
+
+To avoid big property, we need wrapperd decider synthesized by multiple deciders.
+However developers don't want to write specific decider and predicate.
```
However, developers would not want to write specific deciders and predicates because ....
> why?

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+#### Why compile?
+
+To avoid big property, we need wrapperd decider synthesized by multiple deciders.
```
To avoid big property, we need a wrapped decider synthesized by multiple deciders.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+To avoid big property, we need wrapperd decider synthesized by multiple deciders.
+However developers don't want to write specific decider and predicate.
+So, OVM compiler should compile Properties with free variables.
```
OVM compiler, hence, should compile Properties with free variables.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+#### Custome generator
+
+If you wanna support new L1 protocol, you can develop specific generators following [generator interface](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/generator/src/CodeGenerator.ts) and [abstract syntax tree](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/transpiler/src/CompiledPredicate.ts) of DSL.
```
If you want to support a new Layer1 protocol, you can develop specific generators following [generator interface](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/generator/src/CodeGenerator.ts) and [abstract syntax tree](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/transpiler/src/CompiledPredicate.ts) of DSL.

yuriko627 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+#### Custome generator
+
+If you wanna support new L1 protocol, you can develop specific generators following [generator interface](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/generator/src/CodeGenerator.ts) and [abstract syntax tree](https://github.com/cryptoeconomicslab/ovm-compiler/blob/master/packages/transpiler/src/CompiledPredicate.ts) of DSL.
+We can show 2 examples of generator in https://github.com/cryptoeconomicslab/ovm-compiler/tree/master/packages, solidity-generator and ethereum-generator.(ethereum-generator generates EVM bytecode).
```
We can demonstrate two examples of the generator in https://github.com/cryptoeconomicslab/ovm-compiler/tree/master/packages, Solidity-generator and Ethereum-generator, which generates EVM bytecode.

syuhei176 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+### Who should read this document
+
+This document is written primarily for developers who will use our OVM core framework.
```
That's true. What I thought is as follows.
OVM core framework is `wakkanay` it's different from CEL's Plasma framework. for example, Stake Technologies is using `wakkanay` to develop `wakkanay-substrate`.

I will be looking for more appropriate naming than "OVM core framework"!

tkmct replied:
sounds good to me.

syuhei176 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+
+This document is written primarily for developers who will use our OVM core framework.
+
+- Developers who wants to contribute to develop our OVM client should read SDK and framework spec described here (link).
```
fixed!
I removed sentence because this "framework" is core library.

syuhei176 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+# OVM Plasma Chamber Core Spec
```
I'll use [framework name]

syuhei176 commented on core-spec/index.md:
```diff
@@ -0,0 +1,396 @@
+### CEL's framework
+
+Since OVM is concept we had to implement concrete system to benefit from OVM's advantage.
+CEL's framework is making OVM real by some components, OVM contract, OVM client and compiler.
```
thanks! fixed.

syuhei176 commented:
> Awesome work! Thanks for helping me to understand the whole system better. ☺️

@yuriko627 Thank you. I fixed or reply all comments and request changes!
And I'm sorry for poor English. I applied Grammarly check again and fixed some sentences.

yuriko627 commented on core-spec/index.md:
```diff
@@ -207,19 +207,19 @@ Claims property and instantiate new challenge game.
 ### decideClaimToTrue
 
-Set the game decision true whose dispute period has passed.
+Sets the game decision true whose dispute period has already passed.
```
Sets the game decision true when its dispute period has already passed.

yuriko627 commented on core-spec/index.md:
```diff
@@ -207,19 +207,19 @@ Claims property and instantiate new challenge game.
 ### decideClaimToFalse
 
-Set the game decision false when its challenge is set true.
+Set the game decision false when its challenge has been evaluated to true.
```
Set -> Sets 
Sorry for missing to correct here.

yuriko627 commented on core-spec/index.md:
```diff
@@ -207,19 +207,19 @@ Claims property and instantiate new challenge game.
 ### removeChallenge
 
-Removes challenge whose decision is set as false.
+Removes challenge whose decision has been evaluated to false.
```
Removes a challenge when its decision has been evaluated to false.

yuriko627 commented on core-spec/index.md:
```diff
@@ -101,7 +101,7 @@ L1 adaptor must implements Encoder and Decoder for following primitive types.
 ## Wallet
 
-Wallet's responsibility is key management. Wallet provides L1 specific private key management ways.
+Wallet's responsibility is end-users' private key management. Wallet provides a specific way to manage private keys depending upon connected mainchain.
```
Wallet's responsibility is end-users' private key management. Wallet provides a specific way to manage private keys depending on the connected mainchain.
> This just sounds better to me.

syuhei176 replied:
> end-users' private key management.

This is much informative! thanks!

tkmct commented on core-spec/index.md:
```diff
@@ -0,0 +1,400 @@
+### CEL's framework
+
+Since OVM was originally proposed as a concept of shared language for various Layer2 solutions, we have been focusing to implement a concrete system that realizes OVM's unique advantages into real software. This lead to the advent of [framework name].
+When you break [framework name] down, it has some major components, OVM contract, OVM client and a compiler.
```
What about `[framework name] consists of three main components: OVM contract, OVM client and compiler.`

tkmct commented on core-spec/index.md:
```diff
@@ -0,0 +1,400 @@
+
+### CEL's framework
+
+Since OVM was originally proposed as a concept of shared language for various Layer2 solutions, we have been focusing to implement a concrete system that realizes OVM's unique advantages into real software. This lead to the advent of [framework name].
```
I prefer Layer2 protocols over Layer2 solutions. what do you think?

syuhei176 replied:
yes, sounds good!

tkmct commented on core-spec/index.md:
```diff
@@ -0,0 +1,400 @@
+
+### CEL's framework
+
+Since OVM was originally proposed as a concept of shared language for various Layer2 solutions, we have been focusing to implement a concrete system that realizes OVM's unique advantages into real software. This lead to the advent of [framework name].
```
focusing to implement => focusing on implementing

tkmct replied:
a concrete system that realizes OVM's unique advantages into real software => a concrete system to utilize unique advantages of OVM.

tkmct commented on core-spec/index.md:
```diff
@@ -0,0 +1,400 @@
+
+## Overview
+
+OVM is a concept of shared language to standardize various L2 dispute logic. Dapps developers will only have to write property to define dispute logic to develop a scalable dapp using Layer 2 protocols.
```
What about `OVM is abstraction of L2 dispute logic.` instead of `OVM is a concept of shared language to standardize various L2 dispute logic.`?

syuhei176 replied:
good!
